lib/Runtime/Library/JavascriptGenerator.cpp

```diff
@@ -231,7 +231,7 @@ Var JavascriptGenerator::EntryNext(RecyclableObject* function, CallInfo callInfo
 
     AUTO_TAG_NATIVE_LIBRARY_ENTRY(function, callInfo, _u("Generator.prototype.next"));
 
-    if (!VarIs<JavascriptGenerator>(args[0]))
```

Comment:
Looks like it is related to handling of `this`, but I am not sure how exactly it would affect the behavior..

Reply:
Indeed args[0] is this.

This check is for if someone does:
Generator.prototype.next.call(notAGenerator)

The change was needed because VarIs<JavascriptGenerator> returns true for AsyncGenerators as well as normal ones - and this needs to error if it’s being called on an AsyncGenerator - which it now will.